### ISSUE TITLE
Feature request / question: automatic collection of doc pages entry

### ISSUE THREAD

**Issue by charleskawczynski** [fredrikekre/Literate.jl#103]
This is more of a request / question. I'd like to know if this [added feature](https://github.com/climate-machine/CLIMA/pull/1006/commits/940a92b25e83bff035077c840fd711ec079380b3#diff-6afdb61592b3ac8012fbef1b68d7f575) is something that could be incorporated into Literate, or if it's better for Documenter, or just create a separate package.

The gist of the feature (`pages_array`) is that the nested `pages` array of `Pair`'s is automatically determined from a given directory tree with `.jl` literate tutorials. For example, assuming a `literate_examples` folder in the package, the generated markdown files are grabbed from
```julia
# After generating markdown files in `generated_dir` directory
tutorials_generated = [joinpath(root, f) for (root, dirs, files) in Base.Filesystem.walkdir(generated_dir) for f in files]
tutorials_md = filter(x -> endswith(x, ".md"), tutorials_generated)
tutorials_md = map(x -> String(last(split(x, joinpath("docs", "src")))), tutorials_md)

tutorials = pages_array(tutorials_md) # Main feature function

pages = Any[
    "Home" => "index.md",
    "Installation" => "Installation.md",
    "Tutorials" => tutorials,
    ...
]
```

**Comment by johnnychen94** [comment]
FYI, I've made a package DemoCards that handles a similar task. A live example can be found at https://juliaimages.org/latest/democards/examples/

**Comment by charleskawczynski** [comment]
Thanks for the info! I may just make this a separate package since I'd like to keep a light set of dependencies. I'll do that for now and see if it can be incorporated here or in Documenter.

**Comment by fredrikekre** [comment]
This seems like a common enough pattern that I also use myself so might make sense to add the functionality here.

**Comment by fredrikekre** [comment]
Of course, you can already compose it pretty easily with broadcasting:
```
pages = readdir("tutorials-src"; join=true)
Literate.markdown.(pages, out)
```

**Comment by charleskawczynski** [comment]
Ok, thanks. I've added some tests and useful functionality. Feel free to grab/modify whatever from [AutoPages.jl](https://github.com/charleskawczynski/AutoPages.jl/).
